### PR TITLE
feat(mo-doc): nested modules

### LINKED ISSUE #1727: [mo-doc] Documentation isn't generated for nested modules
Shouldn't be too tricky to do, we can probably follow how classes nest declarations.

### PR DESCRIPTION
**Pull request by rvanasa**
Renders nested modules in the HTML, ADoc, and Markdown output formats. 

Fixes #1727.

### CODE REVIEW

File: src/docs/extract.ml [hunk 1]
```diff
@@ -12,6 +12,7 @@ and declaration_doc =
   | Value of value_doc
   | Type of type_doc
   | Class of class_doc
+  | Module of module_doc
```
Comment (crusso) [inline src/docs/extract.ml:15]: You could consider adding a `sort : Type.obj_sort` (object|module|actor) and then using the sort field of `ObjBlockE` below to populate it. Then you could emit doc for objects and actors too.

File: src/docs/extract.ml [hunk 2]
```diff
@@ -133,8 +136,19 @@ struct
         {
           it = Syntax.LetD ({ it = Syntax.VarP { it = name; _ }; _ }, rhs, _);
           _;
-        } ->
-        Some (mk_xref (Xref.XValue name), extract_let_doc rhs name)
+        } -> (
+        match rhs with
+        | Source.{ it = Syntax.ObjBlockE (sort, fields); _ } ->
```
Comment (crusso) [inline src/docs/extract.ml:141]: This is the sort field you want to use. You can assert on sort Type.Memory, which is a hacky label only used to describe stable variable records internally.

**Review by crusso — APPROVED** [review]
Nice.

I think this can go in as is, but you could also easily generalize it to handle nested object declarations and actors (if we ever support nested actor on the IC rather than just in the interpreter). 

If it goes in as is, maybe just open an issue to extend this PR in future for whoever wants to tackle it.

File: src/docs/extract.ml (hunk 1 above)
Comment (rvanasa) [reply src/docs/extract.ml]: Gotcha; thanks. I was 50/50 on generalizing this for the initial PR, but I went ahead and set everything up (along with a few improvements for module and actor classes).

File: src/docs/plain.ml [hunk 3]
```diff
@@ -77,6 +77,16 @@ let plain_of_func_sort : Buffer.t -> Syntax.func_sort -> unit =
...
 
+let plain_of_obj_sort_title : Buffer.t -> Syntax.obj_sort -> unit =
+ fun buf sort ->
+  Buffer.add_string buf
+    Mo_types.Type.(
+      match sort.it with
+      | Object -> "Object "
+      | Actor -> "Actor "
+      | Module -> "Module "
+      | Memory -> "Memory ")
```
Comment (crusso) [inline src/docs/plain.ml:88]:
```suggestion
      | Memory -> "Memory ")
```
This case shouldn't really arise but fine to support it.

**Review by crusso — APPROVED** [review]
Still LGTM. Thanks for  generalizing!

File: src/docs/plain.ml (hunk 3 above)
Comment (rvanasa) [reply src/docs/plain.ml:88]: Makes sense. For some reason this was already included, so I just kept the original logic.

**Pull request merged**